src/components/buttons/critical-action-button/critical-action-button.js

```diff
@@ -0,0 +1,88 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import filterAriaAttributes from '../../../utils/filter-aria-attributes';
+import Spacings from '../../spacings';
+import styles from './critical-action-button.mod.css';
+
+const getStateClassName = ({ isToggled, isDisabled }) => {
+  if (isDisabled) return styles.disabled;
+  if (isToggled) return styles.active;
+  return styles.button;
+};
+const getSizeClassName = size => styles[size];
+
+const CriticalActionButton = props => {
+  const dataProps = {
+    'data-track-component': 'CriticalActionButton',
+    ...filterDataAttributes(props),
+  };
+  const ariaProps = {
+    // TODO: Turn `aria-label` into its own prop and make it required
+    // in order to make the component more accessible. Currently, for backwards
+    // compatibility reasons, `aria-label` contains the `label` prop value.
+    'aria-label': props.label,
+    'aria-disabled': props.isDisabled,
+    'aria-pressed': props.isToggled,
+    ...filterAriaAttributes(props),
+  };
+
+  return (
+    <button
+      onClick={props.onClick}
+      disabled={props.isDisabled}
+      className={styles.reset}
+      type={props.type}
+      {...dataProps}
+      {...ariaProps}
+    >
+      <div
+        className={classnames(
+          getStateClassName({
+            isToggled: props.isToggled,
+            isDisabled: props.isDisabled,
+          }),
+          getSizeClassName(props.size)
+        )}
+      >
+        <Spacings.Inline scale="xs" alignItems="center">
+          {Boolean(props.iconLeft) &&
+            React.cloneElement(props.iconLeft, {
+              theme: props.isDisabled ? 'grey' : 'white',
+            })}
+          <span>{props.label}</span>
+        </Spacings.Inline>
+      </div>
+    </button>
+  );
+};
+
+CriticalActionButton.propTypes = {
+  id: PropTypes.string,
+  name: PropTypes.string,
+  label: PropTypes.string.isRequired,
+  iconLeft: PropTypes.node,
+  isToggled: PropTypes.bool,
+  isDisabled: PropTypes.bool,
+  onClick: PropTypes.func.isRequired,
+  size: PropTypes.oneOf(['big', 'small']),
+  type: PropTypes.oneOf(['button', 'reset', 'submit']),
+
+  // DEPRECATED
+  isToggleButton(props, propName, componentName, ...rest) {
```

Comment:
The idea is to avoid showing business logic through button states. We've introduced `ToggleBar` for that. We still have the `isToggled` to show application logic (if a `SecondaryActionButton` get clicked to display some filters list for example, and clicking it again will hide them)

Reply:
I don't get the vague difference of "business logic" and "application logic".  In the end it's still a button that either supports toggling or not, it seems. But maybe the difference is that now all buttons can potentially be toggled, but they will always look the same when they are not toggled (no matter whether they support toggling or not). I'm fine with this then 👍

Reply:
> I don't get the vague difference of "business logic" and "application logic". In the end it's still a button that either supports toggling or not, it seems.

I'd say that **business logic** is all of the things that belongs to the client's business. Lets take an example of wether a discount is active or not. That is not up to the application's frontend, that belongs to our customer's store logic. 

The **application logic**, is more of how we take the decision to show things in the application (e.g UX decisions on the dos and dont's). A button is supposed to execute an action, and bringing the abstraction of showing a business logic together with it (remember the example of showing an active discount) just makes the component and the user experience more complex. That's why we segregated the logic for business logic for the `ToggleBar` and we keep the buttons to be as simple as to call actions or toggle internal feature states.

I hope the difference is more clear now

> But maybe the difference is that now all buttons can potentially be toggled, but they will always look the same when they are not toggled (no matter whether they support toggling or not)

Every default state of a button looks different in terms of color, because they mean different things (one is a primary action, the other is a critical action etc). The "toggle/active" state is a visual representation of a real-world state when a button is wether pressed or not. Some design approaches deal with it by for example adding inner shadows (to create some 3D spatial simulation just like in real life), others darkens a color etc. I think every button (if its a button) should support that.